lib/hanami/model/mapping.rb

```diff
@@ -1,47 +1,29 @@
-require 'hanami/model/mapping/collection'
-require 'hanami/model/mapping/collection_coercer'
-require 'hanami/model/mapping/coercers'
+require 'transproc'
 
 module Hanami
   module Model
-    # Mapping internal utilities
+    # Mapping
     #
     # @since 0.1.0
-    module Mapping
-      # Unmapped collection error.
-      #
-      # It gets raised when the application tries to access to a non-mapped
-      # collection.
-      #
-      # @since 0.1.0
-      class UnmappedCollectionError < Hanami::Model::Error
-        def initialize(name)
-          super("Cannot find collection: #{ name }")
-        end
+    class Mapping
+      def initialize(&blk)
+        @attributes = {}
+        instance_eval(&blk)
+        @processor = @attributes.empty? ? ::Hash : Transproc(:rename_keys, @attributes)
       end
 
-      # Invalid entity error.
-      #
-      # It gets raised when the application tries to access to a existing
-      # entity.
-      #
-      # @since 0.2.0
-      class EntityNotFound < Hanami::Model::Error
-        def initialize(name)
-          super("Cannot find class for entity: #{ name }")
-        end
+      def model(entity)
       end
 
-      # Invalid repository error.
-      #
-      # It gets raised when the application tries to access to a existing
-      # repository.
-      #
-      # @since 0.2.0
-      class RepositoryNotFound < Hanami::Model::Error
-        def initialize(name)
-          super("Cannot find class for repository: #{ name }")
-        end
+      def register_as(name)
+      end
```

Comment:
The `associations` block is passed as it is to ROM. At this point we're just interested to capture `has_many` calls and discard the rest.